js/invoice/invoiceController.js

```diff
@@ -0,0 +1,110 @@
+firmApp.controller('invoiceCtrl', function ($scope, $http, $uibModal, $filter, localModelInvoice, localRepositoryInvoice) {
+
+//    localRepositoryProduct.getDataInvoice (function(data){
+//        $scope.invoiceList = data;
+//    });
+
+    $scope.propertyName = 'numberInvoice';
+    $scope.reverse = true;
+
+    $scope.sortBy = function(propertyName) {
+        $scope.reverse = ($scope.propertyName === propertyName) ? !$scope.reverse : false;
+        $scope.propertyName = propertyName;
+    };
+
+
+    $scope.addDataInvoice = function(){
+        openDialogAddInvoice();
+    };
+
+    function openDialogAddInvoice(){
+        var modalInstance = $uibModal.open({
+            animation: 'true',
+            templateUrl: 'addInvoiceModal.html',
+            controller: 'addInvoiceCtrl',
+            size: 'lg'
+        });
+
+        modalInstance.result.then(function (newInvoice){
+            if(newInvoice){
+                if($scope.invoiceList){
+                    $scope.invoiceList.push(newInvoice);
+                } else {
+                    $scope.invoiceList = [];
+                    $scope.invoiceList.push(newInvoice);
```

Comment:
to co powyżej jest poprawne, ale można to napiasć znacznie prościej. Zauważ ze w obu przypadkach masz $scope.invoiceList.push(newInvoice); więć najlepiej to wyrzucić poza ifa. Wtedy robi ci się po prostu:
```
if(! $scope.invoiceList)
            $scope.invoiceList = [];

$scope.invoiceList.push(newInvoice);
```



zobacz ile linii kodu mniej :)